### PR TITLE
Add Register and GetAttribute to kmip20

### PR DESCRIPTION
**Pull request by BoonKhoonSim**
1) Add Register and GetAttribute to kmip20 (not validated yet)
2) Rename RegisterKey to Register for kmip14 and kmip20
3) Fix wrong Register parameters used in  clientapi.go

### CODE REVIEW

File: kmip20/op_register.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			return nil, kmip.WithResultReason(merry.UserErrorf("Object Type %s does not match type of cryptographic object provided", payload.ObjectType.String()), kmip14.ResultReasonInvalidField)
+		}
+	}
+
+	respPayload, err := h.RegisterFunc(ctx, &payload)
+	if err != nil {
+		return nil, err
+	}
+
+	req.IDPlaceholder = respPayload.UniqueIdentifier
```
Comment (jskazinski) [inline kmip20/op_register.go]: req.IDPlaceholder = respPayload.UniqueIdentifier
Is this line needed, or even correct? I am not sure why this function is modifying the request object passed in.

File: src/kmipapi/kmip20.go [hunk 2]
```diff
@@ -291,12 +291,144 @@ func (kmips *kmip20service) RevokeKey(ctx context.Context, settings *Configurati
 }
 
 // Register:
-func (kmips *kmip20service) RegisterKey(ctx context.Context, settings *ConfigurationSettings, req *RegisterKeyRequest) (*RegisterKeyResponse, error) {
-	return &RegisterKeyResponse{}, fmt.Errorf("ERROR command is not implemented")
+func (kmips *kmip20service) Register(ctx context.Context, settings *ConfigurationSettings, req *RegisterRequest) (*RegisterResponse, error) {
+	//return &RegisterResponse{}, fmt.Errorf("ERROR command is not implemented")
```
Comment (jskazinski) [inline src/kmipapi/kmip20.go]:
```suggestion
```

File: src/kmipapi/kmip20.go [hunk 3]
```diff
@@ -291,12 +291,144 @@ func (kmips *kmip20service) RevokeKey(ctx context.Context, settings *Configurati
...
+		    },
+	    },
+    }
+
+	var attributes createReqAttrs
+
+	//attributes.ObjectGroup = ObjectGroup{
+	//    AttributeName: "Object Group",
+	//    AttributeValue: "SASED-M-2-14-group",
+    //}
```
Comment (jskazinski) [inline src/kmipapi/kmip20.go]:
```suggestion
```

**Review by jskazinski — APPROVED** [review]
Looks very good. I have one comment about about removing a line and 2 suggestions to remove older commented lines.

File: kmip20/op_register.go (hunk 1 above)
Comment (BoonKhoonSim) [reply kmip20/op_register.go]: This line had been removed in latest checkin.

**Pull request merged**